test/utils.go

```diff
@@ -4,27 +4,29 @@
 package test
 
 import (
+	"fmt"
 	"log"
+	"math/rand"
 	"net"
 	"net/http"
 	"net/http/httptest"
 	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
 func TmpFile(t *testing.T, content string) string {
-	f, err := os.CreateTemp(t.TempDir(), "test")
-	if err != nil {
-		t.Fatal(err)
-	}
-	_, err = f.WriteString(content)
+	dir := t.TempDir()
+	name := filepath.Join(dir, fmt.Sprintf("test%d", rand.Int()))
+
+	// Write content directly without keeping file handle open
+	err := os.WriteFile(name, []byte(content), 0644)
 	if err != nil {
 		t.Fatal(err)
 	}
-	name := f.Name()
-	t.Cleanup(func() { os.RemoveAll(name) })
+
```

Comment:
Can you explain why this change was made?  I see the comment that it won't keep the file handle open -- can you explain why that is needed?  It looks like you removed the cleanup function; does this mean that the temp file will not be removed when the test is over?